src/test/java/io/github/genomicdatainfrastructure/discovery/filters/application/usecases/RetrieveFiltersQueryTest.java

```diff
@@ -0,0 +1,151 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.discovery.filters.application.usecases;
+
+import io.github.genomicdatainfrastructure.discovery.filters.application.ports.FilterBuilder;
+import io.github.genomicdatainfrastructure.discovery.model.Filter;
+import io.github.genomicdatainfrastructure.discovery.model.FilterType;
+import io.github.genomicdatainfrastructure.discovery.model.ValueLabel;
+import jakarta.enterprise.inject.Instance;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class RetrieveFiltersQueryTest {
+
+    @Mock
+    private Instance<FilterBuilder> filterBuilders;
+
+    @Mock
+    private FilterBuilder filterBuilderCkan;
+
+    @Mock
+    private FilterBuilder filterBuilderBeacon;
+
+    @InjectMocks
+    private RetrieveFiltersQuery query;
+
+    @BeforeEach
+    public void setUp() {
+        when(filterBuilders.stream()).thenReturn(Stream.of(filterBuilderCkan, filterBuilderBeacon));
+    }
+
+    @Test
+    void shouldRetrieveAllFilters() {
+        var mockCkanFilters = List.of(
+                Filter.builder()
+                        .source("ckan")
+                        .type(FilterType.DROPDOWN)
+                        .key("tags")
+                        .label("tags")
+                        .values(
+                                List.of(ValueLabel.builder()
+                                        .label("genomics")
+                                        .value("5")
+                                        .build()))
+                        .build());
+
+        when(filterBuilderCkan.build(anyString())).thenReturn(mockCkanFilters);
+
+        var mockBeaconFilters = List.of(
+                Filter.builder()
+                        .source("beacon")
+                        .type(FilterType.DROPDOWN)
+                        .key("Human Phenotype Ontology")
+                        .label("ontology")
+                        .values(
+                                List.of(ValueLabel.builder()
+                                        .label("Motor delay")
+                                        .value("3")
+                                        .build()))
+                        .build());
+        when(filterBuilderBeacon.build(anyString())).thenReturn(mockBeaconFilters);
+
+        var filters = query.execute("token");
+
+        Assert.assertNotNull(filters);
+        Assert.assertEquals(2, filters.size());
```

Comment:
suggestion: you could use `Assertions.assertThat()` instead.

Multiple assert lines is verbose, with the function I mentioned, you can check directly the content of the collection, like `containsExactlyInAnyOrder`, or something like that.